esphome/components/total_daily_energy/sensor.py

```diff
@@ -11,13 +19,26 @@
     "TotalDailyEnergy", sensor.Sensor, cg.Component
 )
 
-CONFIG_SCHEMA = sensor.SENSOR_SCHEMA.extend(
-    {
-        cv.GenerateID(): cv.declare_id(TotalDailyEnergy),
-        cv.GenerateID(CONF_TIME_ID): cv.use_id(time.RealTimeClock),
-        cv.Required(CONF_POWER_ID): cv.use_id(sensor.Sensor),
-    }
-).extend(cv.COMPONENT_SCHEMA)
+CONFIG_SCHEMA = (
+    cv.Schema(
+        {
+            cv.GenerateID(): cv.declare_id(TotalDailyEnergy),
+            cv.GenerateID(CONF_TIME_ID): cv.use_id(time.RealTimeClock),
+            cv.Required(CONF_POWER_ID): cv.use_id(sensor.Sensor),
+        }
+    )
+    .extend(
+        sensor.sensor_schema(
+            UNIT_WATT_HOURS,
```

Comment:
```suggestion
            UNIT_EMPTY,
```

This integration reports the unit based on what's reported by the base sensor: https://github.com/esphome/esphome/blob/8e75980ebd41f1942089b79684bf08aea95232b6/esphome/components/total_daily_energy/total_daily_energy.h#L18-L18

With UNIT_EMPTY the optional won't be set, and the overridden method will be used.

Reply:
Nice, did not know this